parlai/crowdsourcing/utils/analysis.py

```diff
@@ -46,6 +67,27 @@ def compile_results(self) -> pd.DataFrame:
         Each row of the dataframe consists of one utterance of one conversation.
         """
 
+    def compile_and_save_results(self):
+        """
+        Compile results and save them.
+
+        Results will be saved in the format given by --results-format.
```

Comment:
Just a question: does this mean that it is a good design choice to override this and just implement json version in the child class.

Reply:
Hmm, it depends - would your JSON implementation be something that other analysis scripts can use? If so, I'd put it in one of the higher-level abstract compiler classes, but if not, yeah, maybe it'd make sense to put it in the child class